### PR TITLE
Evaluate coefficients columns in pickles

### PR DESCRIPTION
**Pull request by mrmr1993**
This is the Mina counterpart to https://github.com/o1-labs/proof-systems/pull/820.

Still to do:
* [x] fix compilation of WASM bindings (and update JS bindings to handle the new field).

### CODE REVIEW

**Comment by mrmr1993** [comment]
!ci-build-me

File: src/lib/pickles/evaluation_lengths.ml [hunk 1]
```diff
@@ -6,6 +6,7 @@ open Plonk_types
 let create (type a) ~(of_int : int -> a) : a Plonk_types.Evals.t =
   let one = of_int 1 in
   { w = Vector.init Columns.n ~f:(fun _ -> one)
+  ; coefficients = Vector.init Columns.n ~f:(fun _ -> one)
```
Comment (mimoo) [inline src/lib/pickles/evaluation_lengths.ml:9]: I guess same comment here, it'd be good to introduce a different constant for coefficients

File: src/lib/pickles/pickles.ml [hunk 2]
```diff
@@ -1154,6 +1154,8 @@ module Make_str (_ : Wire_types.Concrete) = struct
 
       let () = Tick.Keypair.set_urs_info []
 
+      let () = Backtrace.elide := false
```
Comment (mimoo) [inline src/lib/pickles/pickles.ml:1157]: perhaps worthy of a comment in the code? If I read this line I'm not sure what it is meant to do

File: src/lib/pickles/step_verifier.ml [hunk 3]
```diff
@@ -596,15 +596,20 @@ struct
...
               :: Vector.append w_comm
-                   (Vector.map sigma_comm_init ~f:(fun g -> [| g |]))
-                   (snd Plonk_types.(Columns.add Permuts_minus_1.n)) )
+                   (Vector.append (Vector.map m.coefficients_comm ~f:(fun g -> [| g |]))
+                      (Vector.map sigma_comm_init ~f:(fun g -> [| g |]))
+                      (snd Plonk_types.(Columns.add Permuts_minus_1.n)) )
+                   (snd
+                      Plonk_types.(Columns.add
+                         (fst (Columns.add Permuts_minus_1.n)) ) )
+              )
```
Comment (mimoo) [inline src/lib/pickles/step_verifier.ml]: is it worth splitting this in multiple `let ... in` statements? The indentation here makes it really hard to understand how this array is constructed

Comment (mimoo) [reply src/lib/pickles/step_verifier.ml]: I guess we're just appending everything:

```ocaml
let sg = Vector.map sg_old ~f:(fun g -> [| g |]) in
let coeffs_comm = Vector.map m.coefficients_comm ~f:(fun g -> [| g |]) in
let sigma = Vector.map sigma_comm_init ~f:(fun g -> [| g |]) in
let proof1 = snd Plonk_types.(Columns.add Permuts_minus_1.n) in
let coeff_sigma = Vector.append coeffs_comm sigma proof1 in
let proof2 = snd Plonk_types.(Columns.add (fst (Columns.add Permuts_minus_1.n))) in
let w_and_coeff_sigma = Vector.append w_comm coeff_sigma proof2 in
let rest = [| x_hat |] :: [| ft_comm |] :: z_comm :: [| m.generic_comm |] :: [| m.psm_comm |] :: w_and_coeff_sigma in
let proof3 = snd (Wrap_hack.Padded_length.add num_commitments_without_degree_bound) in
Vector.append sg rest proof3
```

I'm wondering why some can be appended with the normal `::` and some with `Vector.append`

**Review by mimoo — APPROVED** [review]
LGTM

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Pull request merged**